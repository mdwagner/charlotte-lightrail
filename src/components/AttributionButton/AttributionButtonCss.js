import styled from 'styled-components/native';

export const AttributionTouchableOpacity = styled.TouchableOpacity`
  marginHorizontal: 4;
  position: relative;
  height: 20;
  width: 20;
`;

export const AttributionIconImage = styled.Image`
  height: 20;
  width: 20;
`;
