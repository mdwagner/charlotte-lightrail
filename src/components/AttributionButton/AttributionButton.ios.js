import React from 'react';
import { ActionSheetIOS, Alert } from 'react-native';
import defaults from 'react-native-user-defaults';
import { AttributionTouchableOpacity, AttributionIconImage } from './AttributionButtonCss';
import { displayLink } from '../../helpers/config';

export default class AttributionButton extends React.Component {

  showActionSheet = () => {
    const options = [
      '© Mapbox',
      '© OpenStreetMap',
      'Mapbox Telemetry',
      'Cancel',
    ];
    const cancelButtonIndex = 3;
    const title = 'Map Credits and Options';

    const setParticipation = willParticipate => defaults.set('MGLMapboxMetricsEnabled', JSON.stringify(willParticipate)).catch(err => console.log(err));
    const participatingMessage = 'You are helping to make OpenStreetMap and Mapbox maps better by contributing anonymous usage data.';
    const notParticipatingMessage = 'You can help make OpenStreetMap and Mapbox maps better by contributing anonymous usage data.';
    const participatingOptions = [
      { text: 'Tell Me More', onPress: () => displayLink('https://www.mapbox.com/telemetry/') },
      { text: 'Stop Participating', onPress: () => setParticipation('0') },
      { text: 'Keep Participating', style: 'cancel', onPress: () => setParticipation('1') },
    ];
    const notParticipatingOptions = [
      { text: 'Tell Me More', onPress: () => displayLink('https://www.mapbox.com/telemetry/') },
      { text: "Don't Participate", onPress: () => setParticipation('0') },
      { text: 'Participate', style: 'cancel', onPress: () => setParticipation('1') },
    ];
    ActionSheetIOS.showActionSheetWithOptions({ options, cancelButtonIndex, title }, (buttonIndex) => {
      switch (buttonIndex) {
        case 0:
          displayLink('https://www.mapbox.com/about/maps/');
          break;
        case 1:
          displayLink('http://www.openstreetmap.org/about/');
          break;
        case 2:
          defaults.get('MGLMapboxMetricsEnabled')
            .then((participating) => {
              if (participating === '1') {
                Alert.alert('Make Mapbox Maps Better', participatingMessage, participatingOptions);
              } else if (participating === '0') {
                Alert.alert('Make Mapbox Maps Better', notParticipatingMessage, notParticipatingOptions);
              } else {
                console.log('Unexpected value for MGLMapboxMetricsEnabled:', participating);
              }
            });
          break;
        default:
          break;
      }
    });
  }

  render() {
    return (
      <AttributionTouchableOpacity onPress={this.showActionSheet}>
        <AttributionIconImage
          // eslint-disable-next-line
          source={require('../../assets/icons/info/info_circle.png')}
        />
      </AttributionTouchableOpacity>
    );
  }
}
