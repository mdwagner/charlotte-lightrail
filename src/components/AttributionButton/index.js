import { Platform } from 'react-native';
import { default as IOS } from './AttributionButton.ios';
import { default as ANDROID } from './AttributionButton.android';

const exportedComponent = Platform.OS === 'ios' ? IOS : ANDROID;
export default exportedComponent;