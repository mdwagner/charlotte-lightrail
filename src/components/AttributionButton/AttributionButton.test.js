import React from 'react';
import renderer from 'react-test-renderer';
import AttributionButton from './AttributionButton.ios';

describe('<AttributionButton />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <AttributionButton />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
