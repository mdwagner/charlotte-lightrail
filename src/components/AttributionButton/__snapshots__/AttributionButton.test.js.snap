exports[`<AttributionButton /> renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 20,
      "marginHorizontal": 4,
      "opacity": 1,
      "position": "relative",
      "width": 20,
    }
  }
  testID={undefined}>
  <Image
    source={1}
    style={
      Array [
        Object {
          "height": 20,
          "width": 20,
        },
        undefined,
      ]
    } />
</View>
`;
